pkg/controller/elasticquota.go

```diff
@@ -0,0 +1,334 @@
+			UpdateFunc: ctrl.eqUpdated,
+			DeleteFunc: ctrl.eqDeleted,
+		},
+	)
+	klog.V(5).Info("Setting up pod event handlers")
+	podInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    ctrl.podAdded,
+			UpdateFunc: ctrl.podUpdated,
+			DeleteFunc: ctrl.podDeleted,
+		},
+	)
+	return ctrl
+}
+
+func (ctrl *ElasticQuotaController) Run(workers int, stopCh <-chan struct{}) {
+	defer runtime.HandleCrash()
+	defer ctrl.eqQueue.ShutDown()
+	// Start the informer factories to begin populating the informer caches
+	klog.V(2).Info("Starting Elastic Quota control loop")
+	// Wait for the caches to be synced before starting workers
+	klog.V(2).Info("Waiting for informer caches to sync")
+	if ok := cache.WaitForCacheSync(stopCh, ctrl.eqListerSynced, ctrl.podListerSynced); !ok {
+		klog.Error("Cannot sync caches")
+		return
+	}
+	klog.V(2).Info("Elastic Quota sync finished")
+	klog.V(5).Infof("Starting %d workers to process elastic quota", workers)
+	// Launch workers to process elastic quota resources
+	for i := 0; i < workers; i++ {
+		go wait.Until(ctrl.run, time.Second, stopCh)
+	}
+	<-stopCh
+	klog.V(2).Info("Shutting down elastic quota workers")
+}
+
+func (ctrl *ElasticQuotaController) run() {
+	for ctrl.sync() {
+	}
+}
+
+// sync deals with one key off the queue.  It returns false when it's time to quit.
+func (ctrl *ElasticQuotaController) sync() bool {
+	keyObj, quit := ctrl.eqQueue.Get()
+	if quit {
+		return false
+	}
+	err := func(obj interface{}) error {
+		defer ctrl.eqQueue.Done(obj)
+		key, ok := obj.(string)
+		if !ok {
+			ctrl.eqQueue.Forget(obj)
+			runtime.HandleError(fmt.Errorf("expected string in workqueue but got %#v", keyObj))
+			return nil
+		}
+		if err := ctrl.syncHandler(key); err != nil {
+			return fmt.Errorf("error syncing elastic quota '%s': %s", key, err.Error())
+		}
+		ctrl.eqQueue.Forget(obj)
+		klog.V(5).Infof("Successfully synced '%s'", key)
+		return nil
+	}(keyObj)
+	if err != nil {
+		runtime.HandleError(err)
+		return true
+	}
+	return true
+}
+
+// syncHandler syncs elastic quota and convert status.used
+func (ctrl *ElasticQuotaController) syncHandler(key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+	// Get the elastic quota resource with this namespace/name
+	eq, err := ctrl.eqLister.ElasticQuotas(namespace).Get(name)
+	if err != nil {
+		// The elastic quota may no longer exist, in which case we stop processing.
+		if apierrs.IsNotFound(err) {
+			_, err = ctrl.eqClient.SchedulingV1alpha1().ElasticQuotas(namespace).Get(context.TODO(),
+				name, metav1.GetOptions{})
+			if err != nil && apierrs.IsNotFound(err) {
+				// elastic quota was deleted in the meantime, ignore.
+				klog.V(3).Infof("elastic quota has been %q deleted", name)
+				return nil
+			}
+			return nil
```

Comment:
Why we return nil if the eq can be get by the client?

Reply:
Sorry, here is my new idea.
if the eq can not be get by the lister but can be get by the client, that means that the eq was added to the work queue by a delete event. We should delete this eq.

Reply:
I think It's better to delete the eq by the delete event. It's easier to understand.

Reply:
Just like this?
```suggestion
	eq, err := ctrl.eqLister.ElasticQuotas(namespace).Get(name)
	if apierrs.IsNotFound(err) {
		klog.V(5).Infof("Elastic quota has been deleted %v", key)
		return nil
	}
	if err != nil {
		klog.V(3).Infof("Unable to retrieve elastic quota %v from store: %v", key, err)
		return err
	}

```